packages/grid/x-data-grid-generator/src/hooks/useQuery.ts

```diff
@@ -0,0 +1,246 @@
+        : parser(filterItem.value);
+    }
+
+    return filterOperator?.getApplyFilterFn({ filterItem, value: parsedValue }, colDef);
+  });
+
+  if (filterModel.linkOperator === GridLinkOperator.Or) {
+    return rows.filter((row: GridRowModel) =>
+      filterModel.items.some((_, index) => {
+        const value = valueGetters[index](row);
+        return filterFunctions[index] === null ? true : filterFunctions[index]({ value });
+      }),
+    );
+  }
+  return rows.filter((row: GridRowModel) =>
+    filterModel.items.every((_, index) => {
+      const value = valueGetters[index](row);
+      return filterFunctions[index] === null ? true : filterFunctions[index]({ value });
+    }),
+  );
+};
+
+/**
+ * Simulates server data loading
+ */
+const loadServerRows = (
+  rows: GridRowModel[],
+  queryOptions: QueryOptions,
+  serverOptions: ServerOptions,
+  columnsWithDefaultColDef: GridColDef[],
+): Promise<FakeServerResponse> => {
+  const { minDelay = 100, maxDelay = 300, useCursorPagination } = serverOptions;
+
+  if (maxDelay < minDelay) {
+    throw new Error('serverOptions.minDelay is larger than serverOptions.maxDelay ');
+  }
+  const delay = Math.random() * (maxDelay - minDelay) + minDelay;
+
+  const { cursor, page = 0, pageSize } = queryOptions;
+
+  let nextCursor;
+  let firstRowIndex;
+  let lastRowIndex;
+
+  let filteredRows = getFilteredRows(rows, queryOptions.filterModel, columnsWithDefaultColDef);
+
+  const rowComparator = getRowComparator(queryOptions.sortModel, columnsWithDefaultColDef);
+  filteredRows = [...filteredRows].sort(rowComparator);
+
+  const rowNumber = filteredRows.length;
+  if (!pageSize) {
+    firstRowIndex = 0;
+    lastRowIndex = filteredRows.length;
+  } else if (useCursorPagination) {
+    firstRowIndex = cursor ? filteredRows.findIndex(({ id }) => id === cursor) : 0;
+    firstRowIndex = Math.max(firstRowIndex, 0); // if cursor not found return 0
+    lastRowIndex = firstRowIndex + pageSize;
+
+    nextCursor = lastRowIndex >= filteredRows.length ? undefined : filteredRows[lastRowIndex].id;
+  } else {
+    firstRowIndex = page * pageSize;
+    lastRowIndex = (page + 1) * pageSize;
+  }
+  const response: FakeServerResponse = {
+    returnedRows: filteredRows.slice(firstRowIndex, lastRowIndex),
+    nextCursor,
+    rowNumber,
+  };
+
+  return new Promise<FakeServerResponse>((resolve) => {
+    setTimeout(() => {
+      resolve(response);
+    }, delay); // simulate network latency
+  });
+};
+
+interface FakeServerResponse {
+  returnedRows: GridRowModel[];
+  nextCursor?: string;
+  rowNumber: number;
+}
+export interface ServerOptions {
+  minDelay: number;
+  maxDelay: number;
+  useCursorPagination?: boolean;
+}
+
+export interface QueryOptions {
+  cursor?: GridRowId;
+  page?: number;
+  pageSize?: number;
+  // TODO: implement the behavior liked to following models
+  filterModel?: GridFilterModel;
+  sortModel?: GridSortModel;
+}
+
+export const serverConfiguration = (
```

Comment:
It needs to be an imperative name since it's a function. AG Grid calls it `createFakeServer`.